code/BTCUtil.c

```diff
@@ -780,6 +780,17 @@ Datas ckdIndicesFromPath(const char *cStr)
     return DTPopDatas(result);
 }
 
+Data privKeyToHdWallet(Data privKey, const char *passphrase)
+{
+    Data seedPhrase = DataCopy("Bitcoin seed", strlen("Bitcoin seed"));
+
+    Data seed = PBKDF2(toMnemonic(privKey).bytes, passphrase);
+
+    Data hash = hmacSha512(seedPhrase, seed);
+
```

Comment:
Before we pull specific indices off `hash`, we should confirm it is at least that big (lest we create a buffer overrun). Add an assert like so `BTCUTILAssert(hash.length == 64);`